rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+# Networked Dot Storage
+
+> or what does decentralization even mean?
+
+## Authors
+
+- [Hannah Howard], [Protocol Labs]
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Abstract
+
+This RFC puts forward a proposal for the broad parameters of "decentralization" in W3S. It defines both the goals for "decentralization" and identifies where resources should be invested first on the path to "decentralization".
+
+## Non-goals
+
+1. Addressing anything related to tokens
+2. W3S corporate structure or nucleation plans
+
+## Background
+
+The current state of the world for web3.storage is as follows:
+
+- Web3.storage UCAN protocols are open and well specified. 
+
+- We've now created a version of these protocols that can be run from any machine-- so anyone can spin up theoretically own version of our service.
+
+- Seperately Web3.storage backs up data to Filecoin, via open protocols.
+
+- We are actively moving towards supporting running on top of underlying storage mechanisms that lack rich event functionality (a.k.a. bucket events)
+
+## Why Go Further
+
+Today, someone who is not the web3.storage team could spin up their own version of web3.storage, using the open source software we've published and the protocols we've specified. No one has done this in practice, but we are increasingly close to the point where it's plausable. Is this "sufficient decentralization"?
+
+One proposed path forward is to not pursue "decentralization" beyond trying to make other nodes a reality, perhaps through incentivization, and leaning heavily into product improvements that differentiate our platform. We could aim for "S3 if AWS invested in open source S3 implementations and made it content addressed".
+
+This is not the path we are going to pursue. It feels important to say that up front.
+
+But before we go deeper into where we are headed, it's worth looking at the implied dichotomy between "decentralization" and product features. That this dichotomy is almost reflexively referred to likely results from failing to appropriately frame "decentralization" in terms of product features. So let's start by framing the product features we want to unlock through decentralization.
+
+## The Network We Want
+
+### No Data Loss
+
+1. Users should be able to specify how many unique copies of their data they want backed up, and the W3S network should handle insuring that many copies remain backed up as long as the user pay to store in W3S.
+
+### Perpetuity
+
+1. Users should be able to store hot data in W3S that will be available for  retrieval until they stop paying for it. They should not have to worry about any single business entity holding their data disappearing. If it does, the data should go somewhere else without their needing to intervene.
```

Comment:
Love this. At some point I'd really like to game out "what would it look like if w3s shut down suddenly" from the perspective of our users. Really love the idea of centering "our users won't be screwed if we go away" and I'm sensitive to the fact that this is in tension with the traditional startup virtue of having a "defensible business" - I think there is going to be pressure from investors to ensure that our users are dependent on us and can't leave and I think we should be ready to push back on that when it comes.

Reply:
If w3s goes away there is no longer an entity user was paying for service, which would mean user stopped paying so data can disappear, is it not ?

I also wonder if entity providing this assurance should be different from the w3s itself ? In my imagination that entity will be cooperating with w3s and other instances of the protocol deployment to ensure that data can be migrated in case one of them is out of business.

Another question that comes in my head is how can data be kept available in case of entity going out of business as there may not be other entity that offers service for the same price.

Reply:
I mean if only there were some sort of decentralized currency system based on a public database of transactions that wasn't controlled by any one entity that we could use.